File: packages/shared/src/types/aiProvider.ts
/** AI Provider types for multi-vendor configuration. */

export interface AIModelConfig {
  model_id: string;
  display_name: string;
  capability_tier: 'fast' | 'standard' | 'advanced';
  max_tokens: number;
  supports_streaming: boolean;
  supports_image: boolean;
}

export type ProviderType = 'anthropic' | 'gemini' | 'openai_compat';

export interface AIProvider {
  id: string;
  name: string;
  provider_type: ProviderType;
  base_url: string;
  api_key_masked: string;
  models: AIModelConfig[];
  is_default: boolean;
  enabled: boolean;
  priority: number;
  created_at: string | null;
  updated_at: string | null;
}

export interface AIProviderCreateInput {
  name: string;
  provider_type: ProviderType;
  base_url: string;
  api_key: string;
  models: AIModelConfig[];
  is_default: boolean;
  enabled: boolean;
  priority: number;
}

export interface AIProviderUpdateInput {
  name?: string;
  provider_type?: ProviderType;
  base_url?: string;
  api_key?: string;
  models?: AIModelConfig[];
  is_default?: boolean;
  enabled?: boolean;
  priority?: number;
}

export interface TestResult {
  success: boolean;
  message: string;
}

export interface AvailableModelInfo {
  provider_id: string;
  provider_name: string;
  provider_type: ProviderType;
  model_id: string;
  display_name: string;
  max_tokens: number;
  supports_streaming: boolean;
  supports_image: boolean;
}

export interface AvailableModelsByTier {
  fast: AvailableModelInfo[];
  standard: AvailableModelInfo[];
  advanced: AvailableModelInfo[];
}

export const PROVIDER_TYPE_LABELS: Record<ProviderType, string> = {
  anthropic: 'Anthropic',
  gemini: 'Gemini',
  openai_compat: 'OpenAI Compatible',
};

export const CAPABILITY_TIER_LABELS: Record<string, string> = {
  fast: 'Fast',
  standard: 'Standard',
  advanced: 'Advanced',
};

export const DEFAULT_BASE_URLS: Record<ProviderType, string> = {
  anthropic: 'https://api.anthropic.com',
  gemini: 'https://generativelanguage.googleapis.com',
  openai_compat: '',
};

File: backend/models/ai_provider.py
"""AIProvider model — stores multi-vendor AI provider configurations."""

from sqlalchemy import Column, String, Text, JSON, Boolean, Integer

from models.base import Base, TimestampMixin


class AIProvider(Base, TimestampMixin):
    """AI provider configuration (Anthropic, Gemini, OpenAI-compatible, etc.)."""

    __tablename__ = "ai_providers"

    id = Column(String(36), primary_key=True)
    name = Column(String(100), nullable=False)
    provider_type = Column(String(30), nullable=False)  # anthropic | gemini | openai_compat
    base_url = Column(String(500), nullable=False, default="")
    api_key = Column(Text, nullable=False, default="")
    models = Column(JSON, nullable=False, default=lambda: [])
    is_default = Column(Boolean, nullable=False, default=False)
    enabled = Column(Boolean, nullable=False, default=True)
    priority = Column(Integer, nullable=False, default=10)
